src/ComponentRegister.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+import {
+  ComponentRegister,
+  LinkComponent,
+  LinkComponentProps
+} from "./ComponentRegister";
+
+const TestLinkComponent: LinkComponent = ({
+  href,
+  children
+}: LinkComponentProps) => <div data-href={href}>{children}</div>;
+
+describe(".init()", () => {
+  it("registers all of the components provided", () => {
+    ComponentRegister.init({
+      components: {
+        Link: TestLinkComponent
+      }
+    });
+
+    expect(ComponentRegister.Link).toStrictEqual(TestLinkComponent);
+  });
+  it("does not alter properties if none are provided", () => {
```

Comment:
```suggestion

  it("does not alter properties if none are provided", () => {
```

Reply:
I think this is me being thrown off by Jest extensions little debug option, which looks like an extra line being overlaid.

Reply:
Ah yes. Watch out for TypeScript also doing that.